instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/cache/BoundedCache.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.cache;
+
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+final class BoundedCache<K, V> implements Cache<K, V> {
```

Comment:
`BoundedCache` -> `MapBackedCache`? (since nothing bounded about this particular class)

Reply:
Seems like a worthwhile rename